openreview/arr/arr.py

```diff
@@ -558,4 +558,96 @@ def set_SAC_ethics_review_process(self, sac_ethics_flag_duedate=None):
         return self.venue.set_SAC_ethics_review_process(sac_ethics_flag_duedate)
 
     def open_reviewer_recommendation_stage(self, start_date=None, due_date=None, total_recommendations=7):
-        return self.venue.open_reviewer_recommendation_stage(start_date, due_date, total_recommendations)
\ No newline at end of file
+        return self.venue.open_reviewer_recommendation_stage(start_date, due_date, total_recommendations)
+    
+    @classmethod
+    def process_commitment_venue(ARR, client, venue_id):
+
+        def add_readers_to_note(note, readers):
+            if readers[0] in note.readers:
+                return
+            
+            domain = note.domain
+            client.post_note_edit(
+                invitation = f'{domain}/-/Edit',
+                readers = [domain],
+                signatures = [domain],
+                writers = [domain],
+                note = openreview.api.Note(
+                    id = note.id,
+                    readers = {
+                        'append': readers
+                    }
+                )            
+            )    
+
+        def create_readers_group(submission):
+            domain = submission.domain
+
+            commitment_readers_group_id = f'{domain}/Submission{submission.number}/Commitment_Readers'
+
+            commitment_readers_group = openreview.tools.get_group(client, commitment_readers_group_id)
+
+            if commitment_readers_group:
+                print(f'Group already exists, add members {venue_id} to it.')
+                client.add_members_to_group(commitment_readers_group_id, [venue_id])
+                return
+
+            print(f'Creating group {commitment_readers_group_id} for submission {submission.number}.')
+            client.post_group_edit(
+                invitation = f'{domain}/-/Edit',
+                readers = [domain],
+                signatures = [domain],
+                writers = [domain],
+                group = openreview.api.Group(
+                    id = commitment_readers_group_id,
+                    signatures = [domain],
+                    writers = [domain],
+                    readers = [domain],
+                    members = [venue_id]
+                )
+            )
+
+        def add_readers_to_arr_submission(submission):
+
+            domain = submission.domain
+            commitment_readers_group_id = f'{domain}/Submission{submission.number}/Commitment_Readers'
+
+            print(f'Add group as reader of the submission')
+            if 'everyone' not in submission.readers:
+                add_readers_to_note(submission, [commitment_readers_group_id])
+
+            print(f'Add group as reader of the submission review and meta reviews')
+            replies = client.get_notes(forum = submission.id)
+
+            for reply in replies:
+
+                if 'Official_Review' in reply.invitations[0]:
```

Comment:
Do you think we should parameterize which replies get revealed?

Reply:
Yes, I can do that. This could be an optional field where the default value is "Official_Review" and "Meta_Review".

I would support invitation names only for now, I know there is some logic to release comments based on readers.